crates/trie/trie/src/proof/mod.rs

```diff
@@ -183,7 +184,7 @@ where
 
 /// Generates storage merkle proofs.
 #[derive(Debug)]
-pub struct StorageProof<T, H> {
+pub struct StorageProof<T, H, K = AddedRemovedKeys> {
```

Comment:
ah I see, this way this works for both, proofretainer and addedremovedkeys

makes sense